src/workerd/server/tests/weakref/index.mjs

```diff
@@ -0,0 +1,62 @@
+let frCounter = 0;
+let weakRefState = { isDereferenced: false, value: null };
+
+const fr = new FinalizationRegistry(() => {
+  ++frCounter;
+});
+
+export default {
+  async fetch(request, env, ctx) {
+    const url = new URL(request.url);
+    const test = url.searchParams.get('test') || 'fr';
+
+    if (test === 'fr') {
+      // Test FinalizationRegistry
+      (function () {
+        fr.register({}, '');
+      })();
```

Comment:
Nit: this can be just a regular block,, but not critical.

```suggestion
      {
        fr.register({}, '');
      }
```

Reply:
From my observations: regular block changes when the object will get GC'd (it will get garbage collected during the next request even with explicit `gc()`, v8 is probably a bit conservative with locally scoped variables). If we want to observe immediate GC and FinalizationRegistry execution, we need a function call (or an IIFE like this one)